src/services/api/features/package-actions/handlers/router.ts

```diff
@@ -0,0 +1,70 @@
+import { response } from "@/shared/lambda-response";
+import { type APIGatewayEvent } from "aws-lambda";
+import { ActionTypes } from "shared-types";
+import { withdrawRaiLambda } from "./withdraw-rai-lambda";
+import { enableRaiWithdrawLambda } from "./enable-rai-withdraw-lambda";
+import { issueRaiLambda } from "./issue-rai-lambda";
+import { respondToRaiLambda } from "./respond-to-rai-lambda";
+import { APIError } from "../services/error-handle-service";
+import { withdrawPackageLambda } from "./withdraw-package-lambda";
+import { isAuthorized } from "@/libs/auth/user";
+import { getPackage } from "@/libs/package/getPackage";
+
+type Routes = Record<
+  ActionTypes,
+  Promise<ReturnType<typeof response> | string | undefined>
+>;
+
+export const router = async (event: APIGatewayEvent) => {
+  if (!event.pathParameters || !event.pathParameters.actionType) {
+    return response({
+      statusCode: 400,
+      body: { message: "Action type path parameter required" },
+    });
+  }
+  if (!event.body) {
+    return response({
+      statusCode: 400,
+      body: { message: "Event body required" },
+    });
+  }
+
+  const body = JSON.parse(event.body);
+
+  const result = await getPackage(body.id);
+  const passedStateAuth = await isAuthorized(event, result._source.state);
+  if (!passedStateAuth)
+    return response({
+      statusCode: 401,
+      body: { message: "Not authorized to view resources from this state" },
+    });
+  if (!result.found)
+    return response({
+      statusCode: 404,
+      body: { message: "No record found for the given id" },
+    });
```

Comment:
This get-result-and-auth pattern happens a lot, I assume. (Is that a correct assumption?) Would it be useful to bake the auth portion into the helper? We could throw the error from the helper and catch in handlers like this so these responses are universal on any endpoint where that auth fails.

Reply:
Yeah, I think it will be very common anytime the state needs to be checked to see if the user has access to it. Something I have started doing is creating different error objects to make them easy to catch when we bubble them up. Example below
```
try {
...handle lambda stuff here
} catch (error: unknown) {
  if(error instanceof AuthError) {
    ....do auth error stuff here
  } else if(error instanceof SeaToolError) {
    ....do seatool error stuff here
  }
}
```
Thoughts on this?

Reply:
💯 Love the idea of codifying our error types and reactions to them this way! Is this something that's able to be functional so handling errors by type logic is isolated and reusable? Not sure if/when these kinds of things differ, so maybe we figure out those borders and design something to accomodate them.